source/common/unified_memory_allocation/CMakeLists.txt

```diff
@@ -1,11 +1,19 @@
 # Copyright (C) 2023 Intel Corporation
 # SPDX-License-Identifier: MIT
 
-add_library(unified_memory_allocation STATIC
+set(UMA_SOURCES
     src/memory_pool.c
     src/memory_provider.c
 )
 
+if(UMA_BUILD_SHARED_LIBRARY)
+    add_library(unified_memory_allocation SHARED
```

Comment:
`message(WARNING, "Unified Memory Allocation is still an early work in progress. There are no API/ABI backward compatibility guarantees. There will be breakages. Do not use the shared library in production software.")`

Or something to that effect.

Reply:
Done.